StyleCop.Analyzers/StyleCop.Analyzers/Helpers/FileHeader.cs

```diff
@@ -0,0 +1,125 @@
+        internal FileHeader(XElement headerXml, int fileHeaderStart, int fileHeaderEnd)
+        {
+            this.headerXml = headerXml;
+            this.fileHeaderStart = fileHeaderStart;
+            this.fileHeaderEnd = fileHeaderEnd;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="FileHeader"/> instance representing a missing file header.
+        /// </summary>
+        /// <value>
+        /// A <see cref="FileHeader"/> instance representing a missing file header.
+        /// </value>
+        internal static FileHeader MissingFileHeader
+        {
+            get
+            {
+                return new FileHeader { IsMissing = true };
+            }
+        }
+
+        /// <summary>
+        /// Gets a <see cref="FileHeader"/> instance representing a missing file header.
+        /// </summary>
+        /// <value>
+        /// A <see cref="FileHeader"/> instance representing a missing file header.
+        /// </value>
+        internal static FileHeader MalformedFileHeader
+        {
+            get
+            {
+                return new FileHeader { IsMalformed = true };
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the file header is missing.
+        /// </summary>
+        /// <value>
+        /// True if the file header is missing.
+        /// </value>
+        internal bool IsMissing { get; private set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the file header contains a properly formatted XML structure.
+        /// </summary>
+        /// <value>
+        /// True if the file header is not properly formatted xml.
+        /// </value>
+        internal bool IsMalformed { get; private set; }
+
+        /// <summary>
+        /// Gets a XML element from the file header with the given tag name.
+        /// </summary>
+        /// <param name="tagName">The tag name for the node.</param>
+        /// <returns>The requested node, or null if the node could not be found.</returns>
+        internal XElement GetElement(string tagName)
+        {
+            return this.headerXml.Descendants().FirstOrDefault(e => e.Name.LocalName.Equals(tagName, StringComparison.Ordinal));
+        }
+
+        /// <summary>
+        /// Gets the location representing the start of the file header.
+        /// </summary>
+        /// <param name="syntaxTree">The syntrax tree to use for generating the location.</param>
+        /// <returns>The location representing the start of the file header.</returns>
+        internal Location GetLocation(SyntaxTree syntaxTree)
+        {
+            if (this.IsMissing || this.IsMalformed)
+            {
+                return Location.Create(syntaxTree, new TextSpan(0, 0));
+            }
+
+            return Location.Create(syntaxTree, TextSpan.FromBounds(this.fileHeaderStart, Math.Min(this.fileHeaderStart + 2, this.fileHeaderEnd)));
```

Comment:
The fileHeaderStart + 2 is intended to limit the squiggles to the first `//` pair of the entire file header. 

The minimal non-malformed file header would be `// <a/>`, so that's definitely more than 2 characters, I'll remove the `Math.Min` stuff.